### PR TITLE
Added support for an `authorization` header

### PR DESCRIPTION
**Pull request by 4lch4**
As I brought up in #19, the current setup of the flat action doesn't support API endpoints that require an `authorization` header.

This Pull Request will add that ability with an `authorization` key that can be added to an HTTP request like so:

```yaml flat.yml
jobs:
  Twitter:
    runs-on: ubuntu-latest
    steps:
      - name: Setup deno
        uses: denolib/setup-deno@v2
      - name: Check out repo
        uses: actions/checkout@v2
      - name: Trending Global Hashtags/Topics
        uses: 4lch4/flat@v3.1.1
        with:
          http_url: https://api.twitter.com/1.1/trends/place.json?id=1
          authorization: "Bearer ${{secrets.TWITTER_API_BEARER_TOKEN}}"
          downloaded_filename: twitter/trending-global-hashtags.json
          postprocess: ./src/processors/Twitter.ts
```

### CODE REVIEW

**Comment by irealva** [comment]
Hi @4lch4, there is a merge conflict here. Would you mind resolving and recommitting?

**Comment by 4lch4** [comment]
> Hi @4lch4, there is a merge conflict here. Would you mind resolving and recommitting?

Not at all! I'll get a new commit pushed up sometime today/tomorrow 😄

File: src/backends/http.ts [hunk 1]
```diff
@@ -9,6 +9,9 @@ export default async function fetchHTTP(config: HTTPConfig): Promise<string> {
     const response = await axios.get(config.http_url, {
       method: 'get',
       responseType: 'stream',
+      headers: {
```
Comment (irealva) [inline src/backends/http.ts:12]: Would you mind checking of config.authorization exists and only then passing in headers? Could be something like this:

```
const auth = { 
  headers: {
    authorization: config.authorization,
  }
};

const headers = config ? auth : {}

const response = await axios.get(config.http_url, {
  method: 'get',
  responseType: 'stream',
  ...headers,
})
```

Comment (4lch4) [reply src/backends/http.ts:12]: Yeah, I was wondering if I should do that. Good point. I'll add that in and push a new commit.

**Comment by irealva** [comment]
By the way, because we are committing the `dist` folder, it's a little different to fix the conflicts. I've found the easiest way is to:

1. Start on your branch PR
2. `git merge main`
3. Then you'll probably have a conflict in the dist/ folder. Just do `yarn dist` to regenerate it
4. `git add *` to add all the new files
5. git commit
6. Push your changes to your branch

**Comment by irealva** [comment]
Replicated with fixes in #33

**Comment by 4lch4** [comment]
Ahhh.. I knew I was forgetting something >.<

Thank you for fixing my PR and merging this, @irealva. Much appreciated!

**Pull request closed without merging**